tests/model_registry/conftest.py

```diff
@@ -289,14 +293,38 @@ def registered_model(request: FixtureRequest, model_registry_client: ModelRegist
 
 @pytest.fixture()
 def model_registry_operator_pod(admin_client: DynamicClient) -> Pod:
-    model_registry_operator_pods = [
-        pod
-        for pod in Pod.get(dyn_client=admin_client, namespace=py_config["applications_namespace"])
-        if re.match(MR_OPERATOR_NAME, pod.name)
-    ]
-    if not model_registry_operator_pods:
+    """Get the model registry operator pod."""
+    try:
+        for pod_list in TimeoutSampler(
+            wait_timeout=60,
+            sleep=5,
+            func=Pod.get,
+            dyn_client=admin_client,
+            namespace=py_config["applications_namespace"],
+        ):
+            for pod in pod_list:
+                if re.match(MR_OPERATOR_NAME, pod.name):
+                    return pod
+    except TimeoutExpiredError:
         raise ResourceNotFoundError("Model registry operator pod not found")
-    return model_registry_operator_pods[0]
+
+
+@pytest.fixture()
+def model_registry_instance_pod(admin_client: DynamicClient) -> Pod:
+    """Get the model registry instance pod."""
+    try:
+        for instance_pod_list in TimeoutSampler(
+            wait_timeout=60,
+            sleep=5,
+            func=get_pods_by_labels,
```

Comment:
If not needed in model_registry_operator_pod(), then you don't need a function. You can directly call Pod.get() with label_selector.